crates/fuel-core-nats/src/lib.rs

```diff
@@ -53,38 +63,37 @@ pub struct Publisher {
 impl Publisher {
     pub async fn new(
         nats_url: &str,
-        nats_nkey: Option<String>,
+        keys: &NatsNkey,
         chain_id: ChainId,
         base_asset_id: AssetId,
         fuel_core_database: CombinedDatabase,
         blocks_subscription: Receiver<
             Arc<dyn Deref<Target = ImportResult> + Send + Sync>,
         >,
     ) -> anyhow::Result<Self> {
+        let seed = keys.seed.clone();
         Ok(Publisher {
             chain_id,
             base_asset_id,
             fuel_core_database,
             blocks_subscription,
-            nats: Self::connect_to_nats(nats_url, nats_nkey).await?,
+            nats: Self::connect_to_nats(nats_url, seed).await?,
         })
     }
 
     async fn connect_to_nats(
         nats_url: &str,
-        nats_nkey: Option<String>,
+        seed_key: Option<String>,
     ) -> anyhow::Result<NatsConnection> {
-        let client = match nats_nkey {
-            Some(nkey) => async_nats::connect_with_options(
-                nats_url,
-                async_nats::ConnectOptions::with_nkey(nkey),
-            )
-            .await
-            .context(format!("Connecting to {nats_url}"))?,
-            None => async_nats::connect(nats_url)
-                .await
-                .context(format!("Connecting to {nats_url}"))?,
-        };
+        if seed_key.is_none() {
+            anyhow::bail!("NATS NKEY seed is empty")
+        }
```

Comment:
If we don't want to ever connect to any `Nats` cluster without the NKey authorization scheme, we can type-safe `seed_key` to be `String` rather than defensively error out at this point.

Reply:
Fwiw, there's a missing test case for invalid `NKey Seed` inputs.